packages/tools/webpack-fluid-loader/src/loader.ts

```diff
@@ -349,14 +407,14 @@ async function attachContainer(
         };
 
         // If we are in side-by-side mode, we need to display the following message in the right div passed here.
-        if (div) {
-            div.innerText = "Waiting for container attach";
+        if (rightDiv) {
+            rightDiv.innerText = "Waiting for container attach";
         }
     } else {
-        await container.attach(attachUrl);
+        await currentContainer.attach(attachUrl);
         replaceUrl();
         attached.resolve();
     }
 
-    return attached.promise;
+    return Promise.all([attached.promise, Promise.resolve(currentContainer)]);
```

Comment:
I think you can replace it with
```
await attached.promise;
return currentContainer;
```
In other words - is there a scenario where someone wants to play with container while it is being attached?
If not, I'd simplify API.

That's said, it's important scenario. It would be nice to inject some big delay into POST such that we can poke at container while it is in attaching state.
Maybe we immediately return container without waiting for attached promise? And ensure that all the code that uses this call is aware of attaching state and can deal with this state (and has right tools to do so)?

Reply:
About the changes during the container attachment, I can think about it but not in this PR. Also we do have several E2E tests for that purpose.

Reply:
E2E tests are great, but exposing this to the user to manually poke at it is as valuable. I thin the ideal solution (if it's possible) that we always delay attachment by couple seconds and just have some indication in UI that it is in flight. Even better solution is the one that exposes more control to user to click buttons and control it even further (i.e. keep it in "attaching" state for minutes). I bet we will find a bunch of bugs that way - here, not in Teams at 100M scale deployment.